app/controllers/api/databases.php

```diff
@@ -1673,11 +1676,54 @@ function updateAttribute(
             throw new Exception(Exception::COLLECTION_NOT_FOUND);
         }
 
-        $attributes = $collection->getAttribute('attributes');
+        $queries = Query::parseQueries($queries);
+        foreach ($queries as $query) {
+            if ($query->getMethod() ===  Query::TYPE_SELECT) {
+                throw new Exception(Exception::GENERAL_QUERY_INVALID, 'Select queries are not valid.');
+            }
+        }
```

Comment:
Why do we disallow select queries?

Reply:
Select queries are not supported due to the limitation of no "key" in the metadata table, which is being used to validate select queries. Also, @TorstenDittmann  said most of the keys are not optional so it's not worth it. 
https://github.com/utopia-php/database/blob/a9f706034a12c7c87e6fd38eca69587fd3def0e0/src/Database/Database.php#L4509

Reply:
What's the error if they're enabled? I think the `key` attribute should be there since it's in the collections config here: https://github.com/appwrite/appwrite/blob/887abad1904936bf9d328443598fec504c01107e/app/config/collections.php#L245

Select queries might not particularly useful for these resources but I would suggest we allow them if easy enough, so we can remove this code and keep the controller complexity to a minimum

Reply:
<img width="596" alt="Screenshot 2023-08-01 at 4 21 21 PM" src="https://github.com/appwrite/appwrite/assets/30731059/f8b6c892-23d4-402e-a282-a67e04c5ea06">
<img width="589" alt="Screenshot 2023-08-01 at 4 20 33 PM" src="https://github.com/appwrite/appwrite/assets/30731059/9f3c8e6e-17f6-4f9e-82a3-4f2c687abe5f">


I think if we use $id instead of $key in validating select queries then we can maybe use that because $id exists in all rows of metadata attributes and indexes

Reply:
We can try get the key and fallback to $id if it's not found, we're doing this a few other places I think

Reply:
Made the changes. But sadly list attributes endpoint will have to be much more complex for  adding select queries support, we need to have type attribute for validating the response model for Attributes list. 

If select query does not include type support, we will have to add type in select query and then remove type property from output.  

discussed two options with @christyjacob4, either we use validator to not support select queries or we support select queries but we add type in select query when it doesn't include it. 

We decided to keep endpont consistent with select queries, we need to add it in select queries.